Inferences/DefinitionIntroduction.hpp

```diff
@@ -26,21 +26,16 @@ class DefinitionIntroduction: public GeneratingInferenceEngine, public Index {
   void attach(SaturationAlgorithm *salg) override {
     GeneratingInferenceEngine::attach(salg);
     attachContainer(salg->getPassiveClauseContainer());
-    reset = false;
   }
 
   void handleClause(Clause *cl, bool adding) override {
-    if(reset)
-      _definitions.reset();
-
-    reset = false;
-    if(adding)
+    if(adding) {
       process(cl);
+    }
   }
 
   ClauseIterator generateClauses(Clause *cl) override {
-    reset = true;
-    return pvi(decltype(_definitions)::Iterator(_definitions));
+    return pvi(ownedArrayishIterator(std::move(_definitions)));
```

Comment:
Does `std::move` do what you want here? I think it's only supposed to leave the moved-from object in some unspecified and possibly-bad state safe for destruction - but maybe we're OK here.

PS I'm glad you know the right `ownerArrayishIterator` magic!

Reply:
This is what Joe does elsewhere, e.g. `CompositeSGI::generateSimplify` (the `move` on `Stack` is done via swap with essentially zeros, so leaves the thing in a sound state).